tests/static/test_strategies/download.py

```diff
@@ -64,3 +65,65 @@ def get(self, session: "Optional[dict[str, Any]]" = None) -> "dict[str, Any]":
                 key = cache.add(handle.read())
 
         return {"key": key}
+
+
+class HTTPSConfig(AttrDict):
+    """HTTP(S)-specific Configuration Data Model."""
+
+    datacache_config: Optional[DataCacheConfig] = Field(
+        None,
+        description=(
+            "Configurations for the data cache for storing the downloaded file "
+            "content."
+        ),
+    )
+
+
+class HTTPSDemoConfig(ResourceConfig):
+    """HTTP(S) download strategy filter config."""
+
+    downloadUrl: AnyHttpUrl = Field(
+        ..., description="The HTTP(S) URL, which will be downloaded."
+    )
+    configuration: HTTPSConfig = Field(
+        HTTPSConfig(), description="HTTP(S) download strategy-specific configuration."
+    )
+
+
+class HTTPDownloadContent(AttrDict):
+    """Class for returning values from Download HTTPS strategy."""
+
+    key: str = Field(..., description="Key to access the data in the cache.")
+
+
+@dataclass
+class HTTPSStrategy:
+    """Strategy for retrieving data via http.
+
+    **Registers strategies**:
+
+    - `("scheme", "http")`
+    - `("scheme", "https")`
+
+    """
+
+    download_config: HTTPSDemoConfig
+
+    def initialize(self) -> AttrDict:
+        """Initialize."""
+        return AttrDict()
+
+    def get(self) -> HTTPDownloadContent:
+        """Download via http/https and store on local cache."""
+        cache = DataCache(self.download_config.configuration.datacache_config)
+        if cache.config.accessKey and cache.config.accessKey in cache:
+            key = cache.config.accessKey
+        else:
+            req = requests.get(
```

Comment:
We really shouldn't request external URLs for tests, we should instead mock the response.